cfpb/consumerfinance.gov
# Apply cache tags to all filterable lists

This PR moves the cache tag implementation from the `Newsroom` to `FilterableListMixin`, and makes the tag the page’s slug. This enables cache tags for all filterable list pages. It also moves the cache purge for newsroom to the general filterable list page invalidation signal handler above to work for all filterable list pages.

While working on cache tagging, it also converts our `CCDBSearchView` into a more generic `CacheTaggedTemplateView` to add cache tags to what is otherwise a normal `TemplateView` defined in `urls.py`.


## How to test this PR

Pull the branch down locally (or use the dev Docker deploy) and run:

```
curl -sI http://localhost:8000/about-us/blog/
```

Observe the `Edge-Cache-Tag: blog` header. 

You can also check the RSS feed URL and other filterable lists and confirm they have `Edge-Cache-Tag` headers as well:

```
curl -sI http://localhost:8000/about-us/blog/feed/
curl -sI http://localhost:8000/about-us/newsroom/
```

You can also check that the CCDB gets the `complaints` cache tag:

```
curl -sI http://localhost:8000/data-research/consumer-complaints/search/
```

## Checklist
